app/controllers/address_controller.rb

```diff
@@ -0,0 +1,35 @@
+class AddressController < ApplicationController
+  def create
+    user = User.find_by(verification_token: params[:verification_token])
+    address = user.addresses.build(permitted_address_params)
+
+    if address.save
+      order = logged_in_user_cart_order
+      order.shipping_address_id = address.id
+      order.save
+      redirect_to order_url(logged_in_user_cart_order), notice: t(:successfull, resource_name: 'added shipping address for order')
+    else
+      redirect_to new_address_url, notice: t(:default_error_message)
+    end
+  end
+
+  def new
+    @address = Address.new
+    @order = Order.find(params[:order_id])
+  end
+
+  def shipping
```

Comment:
how is this action different from the create action above?

Reply:
Shipping is updating the order's shipping address id and handling the 'select from previous addresses' dropdown form submission.
While create is creating a new address, associating it with user and updating order's shipping address id, and handling the 'Create new address' form submission.